### ISSUE TITLE
Add CSRF Tokens for basic security protection

### ISSUE THREAD

**Issue by chris001** [salesagility/SuiteCRM#3209]
https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)_Prevention_Cheat_Sheet#Synchronizer_.28CSRF.29_Tokens

https://developer.sugarcrm.com/2016/11/15/security-changes-coming-in-sugar-7-8/

#### Issue

The SuiteCRM application is weakly protected by HTTP `Referer:` header checking only, which can be easily forged by PDF and other attack vectors, which can then steal information from the CRM and forward it to malicious attacker controlled data-mining servers systems, creating the ultimate nightmare to privacy and competitiveness: Advanced Persistent Threats.

#### Expected Behavior

SuiteCRM needs to add standard baseline protection using CSRF tokens.  Cross Site Request Forgery is prevented.  Each request has a token which is locked to the user browser session.  

#### Actual Behavior

Attackers cannot forge requests by setting the expected referer and steal data by accessing forms.

#### Possible Fix

Add code to implement CSRF tokens on all pages, plus add the related `csrf` settings array in `config.php`.

#### Steps to Reproduce

1.
2.
3.
4.

#### Context

Medium High.  Information security is high priority.

#### Your Environment

* SuiteCRM Version used: latest 7.8.2
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Firefox latest.
* Environment name and version (e.g. MySQL, PHP 7): PHP 5.6 MySQL 5.6
* Operating System and version (e.g Ubuntu 16.04): Linux Debian 8

**Comment by chris001** [comment]
Relatively easily accomplished by adding a few lines of code.  The heavy lifting is done by the vendor package `symfony/security-csrf`

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add CSRF Tokens for basic security protection](https://trello.com/c/IuE65n3O/20-add-csrf-tokens-for-basic-security-protection)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
